simulator/mips/t/unit_test.cpp

```diff
@@ -236,10 +171,621 @@ TEST_CASE( "MIPS32_instr_disasm: Process_Disasm_Branches")
     CHECK(MIPS32Instr(0x06320003).get_disasm() == "bltzall $s1, 3");
     CHECK(MIPS32Instr(0x0633fffb).get_disasm() == "bgezall $s1, -5");
     CHECK(MIPS32Instr(0x0633000a).get_disasm() == "bgezall $s1, 10");
-
     CHECK(MIPS32Instr(0x0bfffb2e).get_disasm() == "j 0x3fffb2e");
     CHECK(MIPS32Instr(0x080004d2).get_disasm() == "j 0x4d2");
     CHECK(MIPS32Instr(0x0ffffb2e).get_disasm() == "jal 0x3fffb2e");
     CHECK(MIPS32Instr(0x0c0004d2).get_disasm() == "jal 0x4d2");
 }
 
+
+// ********* Converted SPIM TT tests **********
+// 27 done
+
+TEST_CASE ( "MIPS32_instr: add")
+{
+    CHECK(MIPS32Instr(0x01398820).get_disasm() == "add $s1, $t1, $t9");
+    MIPS32Instr instr( "add");
+
+    instr.set_v_src( 0, 0);
+    instr.set_v_src( 0, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0);
+
+    instr.set_v_src( 0, 0);
+    instr.set_v_src( 1, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 1);
+
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 0xffffffff, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0);
+
+/*     Overflow exception isn't implemented
```

Comment:
add and addi must trap if overflow occurs, but there are none in implementation.

Reply:
Please refer to the issue #130 in comments.
It should be very simple to implement them with these tests.